### PR TITLE
Upgrade pallets to FRAMEv2

### PR DESCRIPTION
**Pull request by KiChjang**

### CODE REVIEW

**Comment by KiChjang** [comment]
Don't merge yet, I just caught a bug.

**Comment by KiChjang** [comment]
Bug fixed... hopefully.

File: pallets/parachain-system/src/lib.rs [hunk 1]
```diff
@@ -836,919 +405,480 @@ decl_error! {
...
 
-	impl XcmpMessageHandler for SaveIntoThreadLocal {
-		fn handle_xcmp_messages<'a, I: Iterator<Item=(ParaId, RelayBlockNumber, &'a [u8])>>(
-			iter: I,
-			_max_weight: Weight,
-		) -> Weight {
-			HANDLED_XCMP_MESSAGES.with(|m| {
-				for (sender, sent_at, message) in iter {
-					m.borrow_mut().push((sender, sent_at, message.to_vec()));
+	impl<T: Config> GetChannelInfo for Pallet<T> {
```
Comment (thiolliere) [inline pallets/parachain-system/src/lib.rs:513]: in general people tends to put in the pallet module only the items required by the pallet macro.
And they put other stuff outside of the macro.
Though I don't have strong opinion.

File: pallets/parachain-system/src/lib.rs [hunk 2]
```diff
@@ -494,43 +600,43 @@ impl<T: Config> Module<T> {
...
+	/// **Panics** if it turns out that after processing all messages the Message Queue Chain
+	///            hash doesn't match the expected.
 	fn process_inbound_downward_messages(
 		expected_dmq_mqc_head: relay_chain::Hash,
 		downward_messages: Vec<InboundDownwardMessage>,
 	) -> Weight {
 		let dm_count = downward_messages.len() as u32;
 		let mut weight_used = 0;
-
+
```
Comment (thiolliere) [inline pallets/parachain-system/src/lib.rs:631]:
```suggestion

```

File: pallets/xcmp-queue/src/lib.rs [hunk 3]
```diff
@@ -25,29 +25,145 @@
...
+	impl<T: Config> Pallet<T> {}
 
-	/// Information on the avaialble XCMP channels.
-	type ChannelInfo: GetChannelInfo;
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	#[pallet::metadata(T::Hash = "Hash")]
+	pub enum Event<T: Config> {
+		/// Some XCM was executed ok.
+		Success(Option<T::Hash>),
```
Comment (thiolliere) [inline pallets/xcmp-queue/src/lib.rs:96]: I think the metadata will contains the type `"Option<T::Hash>` and not `Option<Hash>` as wanted.

**Review by thiolliere — COMMENTED** [review]
changes are good to me, but master needs to be merged carefully.

Maybe splitting the upgrade per pallet would ease merging master and review but I'm ok anyway

File: pallets/xcmp-queue/src/lib.rs (hunk 3 above)
Comment (KiChjang) [reply pallets/xcmp-queue/src/lib.rs:96]: Wouldn't the attribute above rewrite it to `Option<Hash>`?

**Comment by bkchr** [comment]
@KiChjang please stop using force pushes. Ty

File: pallets/xcmp-queue/src/lib.rs (hunk 3 above)
Comment (thiolliere) [reply pallets/xcmp-queue/src/lib.rs:96]: I don't think so, did you try by comparing metadata before and after the implementation ?
In order to solve it the metadata attribute should contain: `Option<T::Hash> = "Option<Hash>"`

I know we could improve the macro maybe, but I think the metadata will be changed to use scale-info in a not too distant future.

Comment (KiChjang) [reply pallets/xcmp-queue/src/lib.rs:96]: I didn't check specifically on this pallet, but while I was upgrading some other pallets in substrate, type parameters do get changed by the metadata attribute. Will build and check on this pallet as well.

Comment (KiChjang) [reply pallets/xcmp-queue/src/lib.rs:96]: You're right, it doesn't change it, let me push a commit to fix this.

**Review by bkchr — APPROVED** [review]

File: pallets/parachain-system/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,936 @@
...
+	relay_chain::BlockNumber as RelayBlockNumber,
+};
+use cumulus_test_relay_sproof_builder::RelayStateSproofBuilder;
+use frame_support::{
+	assert_ok,
+	dispatch::UnfilteredDispatchable,
+	parameter_types,
+	storage,
+	traits::{OnFinalize, OnInitialize},
+	weights::Weight,
```
Comment (bkchr) [inline pallets/parachain-system/src/tests.rs:31]:
```suggestion
	weights::Weight,
	inherent::{InherentData, ProvideInherent},
```

File: pallets/parachain-system/src/tests.rs [hunk 5]
```diff
@@ -0,0 +1,936 @@
+
```
Comment (bkchr) [inline pallets/parachain-system/src/tests.rs:1]:
```suggestion
```

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge cancel

**Comment by bkchr** [comment]
bot merge

**Pull request merged**